README.md
<p align="center">
  <a href="http://marblejs.com">
    <img src="https://avatars0.githubusercontent.com/u/28954222?s=400&u=cbf28d95c3143ebe7424accb8517110c4d3efde0&v=4" width="320" alt="WaterCodeMelons"/>
  </a>
</p>

<p align="center">
  <a href="https://travis-ci.com/WaterCodeMelons/gamification-front">
    <img src="https://travis-ci.com/WaterCodeMelons/gamification-front.svg?branch=develop" alt="Travis-CI status" height="40">
  </a>
</p>

<p align="center">
Gamification // @ TODO
</p>

## <a name="docs"></a> Documentation

## Contributing

**So, how can I contribute?** 🤔

// @ TODO

- Open a Pull Request and to The Moon we go 🚀

## Authors

<table border="0">
  <tr>
    <td>
      <a href="https://github.com/DawidYerginyan" style="color: white">
        <img src="https://github.com/DawidYerginyan.png?s=150" width="150"/>
      </a>
    </td>
    <td>
      <p><strong>Dawid Yerginyan</strong></p>
      <p><strong>contact: </strong><a href="mailto:dawid.yerginyan@gmail.com">dawid.yerginyan@gmail.com</a></p>
      <p><strong>linkedin: </strong><a href="https://www.linkedin.com/in/dawidyerginyan/">in/dawidyerginyan</a></p>
    </td>
  </tr>
</table>

<table border="0">
  <tr>
    <td>
      <a href="https://github.com/aleksanderujek" style="color: white">
        <img src="https://github.com/aleksanderujek.png?s=150" width="150"/>
      </a>
    </td>
    <td>
      <p><strong>Aleksander Ujek</strong></p>
      <p><strong>contact: </strong><a href="mailto:aleksander.ujek@gmail.com">aleksander.ujek@gmail.com</a></p>
      <p><strong>linkedin: </strong><a href="https://www.linkedin.com/in/aleksanderujek/">in/aleksanderujek</a></p>
    </td>
  </tr>
</table>

<table border="0">
  <tr>
    <td>
      <a href="https://github.com/SebastianWolski" style="color: white">
        <img src="https://github.com/SebastianWolski.png?s=150" width="150"/>
      </a>
    </td>
    <td>
      <p><strong>Sebastian Wolski</strong></p>
      <p><strong>contact: </strong><a href="mailto:sebastian.wolski96@gmail.com">sebastian.wolski96@gmail.com</a></p>
      <p><strong>linkedin: </strong><a href="https://www.linkedin.com/in/sebastian-wolski/">in/sebastian-wolski</a></p>
    </td>
  </tr>
</table>

<table border="0">
  <tr>
    <td>
      <a href="https://github.com/WlodarczykKamil" style="color: white">
        <img src="https://github.com/WlodarczykKamil.png?s=150" width="150"/>
      </a>
    </td>
    <td>
      <p><strong>Kamil Włodarczyk</strong></p>
      <p><strong>contact: </strong><a href="mailto:k.wlodarczyk01@gmail.com">k.wlodarczyk01@gmail.com</a></p>
      <p><strong>linkedin: </strong><a href="https://www.linkedin.com/in/kamilwlodarczyk96/">in/kamilwlodarczyk96</a></p>
    </td>
  </tr>
</table>

## License

MIT licensed


CHANGELOG.md
# Version 0.0.1

Initial project setup

- [NEW] package.json
- [NEW] LICENSE
- [NEW] README.md
- [NEW] CHANGELOG.md
- [NEW] docs/PULL_REQUEST_TEMPLATE.md
- [DEV] `.editorconfig`
- [DEV] `.gitignore`


src/config/serviceWorker.js
// This lets the app load faster on subsequent visits in production, and gives
// it offline capabilities. However, it also means that developers (and users)
// will only see deployed updates on the 'N+1' visit to a page, since previously
// cached resources are updated in the background.

// To learn more about the benefits of this model, read https://goo.gl/KwvDNy.
// This link also includes instructions on opting out of this behavior.

const isLocalhost = Boolean(
  window.location.hostname === 'localhost' ||
  // [::1] is the IPv6 localhost address.
  window.location.hostname === '[::1]' ||
  // 127.0.0.1/8 is considered localhost for IPv4.
  window.location.hostname.match(
    /^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/,
  ),
);

export function register() {
  if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
    window.addEventListener('load', () => {
      const swUrl = 'service-worker.js';

      if (isLocalhost) {
        // This is running on localhost. Lets check if a service worker still exists or not.
        checkValidServiceWorker(swUrl);

        // Add some additional logging to localhost, pointing developers to the
        // service worker/PWA documentation.
        navigator.serviceWorker.ready.then(() => {
          console.log(
            'This web app is being served cache-first by a service ' +
            'worker. To learn more, visit https://goo.gl/SC7cgQ',
          );
        });
      } else {
        // Is not local host. Just register service worker
        registerValidSW(swUrl);
      }
    });
  }
}

function registerValidSW(swUrl) {
  navigator.serviceWorker
    .register(swUrl)
    .then(registration => {
      registration.onupdatefound = () => {
        const installingWorker = registration.installing;
        if (installingWorker) {
          installingWorker.onstatechange = () => {
            if (installingWorker.state === 'installed') {
              if (navigator.serviceWorker.controller) {
                // At this point, the old content will have been purged and
                // the fresh content will have been added to the cache.
                // It's the perfect time to display a 'New content is
                // available; please refresh.' message in your web app.
                console.log('New content is available; please refresh.');
              } else {
                // At this point, everything has been precached.
                // It's the perfect time to display a
                // 'Content is cached for offline use.' message.
                console.log('Content is cached for offline use.');
              }
            }
          };
        }
      };
    })
    .catch(error => {
      console.error('Error during service worker registration:', error);
    });
}

function checkValidServiceWorker(swUrl) {
  // Check if the service worker can be found. If it can't reload the page.
  fetch(swUrl)
    .then(response => {
      // Ensure service worker exists, and that we really are getting a JS file.
      if (
        response.status === 404 ||
        response.headers.get('content-type').indexOf('javascript') === -1
      ) {
        // No service worker found. Probably a different app. Reload the page.
        navigator.serviceWorker.ready.then(registration => {
          registration.unregister().then(() => {
            window.location.reload();
          });
        });
      } else {
        // Service worker found. Proceed as normal.
        registerValidSW(swUrl);
      }
    })
    .catch(() => {
      console.log(
        'No internet connection found. App is running in offline mode.',
      );
    });
}

export function unregister() {
  if ('serviceWorker' in navigator) {
    navigator.serviceWorker.ready.then(registration => {
      registration.unregister();
    });
  }
}


src/config/serviceWorker.ts
// This lets the app load faster on subsequent visits in production, and gives
// it offline capabilities. However, it also means that developers (and users)
// will only see deployed updates on the 'N+1' visit to a page, since previously
// cached resources are updated in the background.

// To learn more about the benefits of this model, read https://goo.gl/KwvDNy.
// This link also includes instructions on opting out of this behavior.

const isLocalhost = Boolean(
  window.location.hostname === 'localhost' ||
    // [::1] is the IPv6 localhost address.
    window.location.hostname === '[::1]' ||
    // 127.0.0.1/8 is considered localhost for IPv4.
    window.location.hostname.match(
      /^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/
    )
);

export function register() {
  if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
    window.addEventListener('load', () => {
      const swUrl = 'service-worker.js';

      if (isLocalhost) {
        // This is running on localhost. Lets check if a service worker still exists or not.
        checkValidServiceWorker(swUrl);

        // Add some additional logging to localhost, pointing developers to the
        // service worker/PWA documentation.
        navigator.serviceWorker.ready.then(() => {
          console.log(
            'This web app is being served cache-first by a service ' +
              'worker. To learn more, visit https://goo.gl/SC7cgQ'
          );
        });
      } else {
        // Is not local host. Just register service worker
        registerValidSW(swUrl);
      }
    });
  }
}

function registerValidSW(swUrl) {
  navigator.serviceWorker
    .register(swUrl)
    .then(registration => {
      registration.onupdatefound = () => {
        const installingWorker = registration.installing;
        if (installingWorker) {
          installingWorker.onstatechange = () => {
            if (installingWorker.state === 'installed') {
              if (navigator.serviceWorker.controller) {
                // At this point, the old content will have been purged and
                // the fresh content will have been added to the cache.
                // It's the perfect time to display a 'New content is
                // available; please refresh.' message in your web app.
                console.log('New content is available; please refresh.');
              } else {
                // At this point, everything has been precached.
                // It's the perfect time to display a
                // 'Content is cached for offline use.' message.
                console.log('Content is cached for offline use.');
              }
            }
          };
        }
      };
    })
    .catch(error => {
      console.error('Error during service worker registration:', error);
    });
}

function checkValidServiceWorker(swUrl) {
  // Check if the service worker can be found. If it can't reload the page.
  fetch(swUrl)
    .then(response => {
      // Ensure service worker exists, and that we really are getting a JS file.
      if (
        response.status === 404 ||
        response.headers.get('content-type').indexOf('javascript') === -1
      ) {
        // No service worker found. Probably a different app. Reload the page.
        navigator.serviceWorker.ready.then(registration => {
          registration.unregister().then(() => {
            window.location.reload();
          });
        });
      } else {
        // Service worker found. Proceed as normal.
        registerValidSW(swUrl);
      }
    })
    .catch(() => {
      console.log(
        'No internet connection found. App is running in offline mode.'
      );
    });
}

export function unregister() {
  if ('serviceWorker' in navigator) {
    navigator.serviceWorker.ready.then(registration => {
      registration.unregister();
    });
  }
}
